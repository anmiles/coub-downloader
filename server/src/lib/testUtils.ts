import type { Coub } from 'types/coub';

function createTestCoub(id: string, override?: Partial<Coub>): Coub {
	return {
		title         : `Title: ${id}`,
		permalink     : id,
		picture       : `${id}.jpg`,
		likes_count   : 1,
		views_count   : 1,
		file_versions : {
			html5 : {
				video : {},
				audio : {},
			},
		},
		media_blocks : {
			external_raw_videos : [],
		},
		...override,
	};
}

export { createTestCoub };
export default { createTestCoub };
