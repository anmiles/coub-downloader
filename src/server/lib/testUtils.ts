export { createTestCoub };
import type { Coub } from '../types/coub';

function createTestCoub(id: string, override?: Partial<Coub>): Coub {
	return {
		permalink: id,
		title: `Title: ${id}`,
		file_versions: {
			html5: {
				video: {},
				audio: {}
			}
		},
		media_blocks: {
			external_raw_videos: [],
		},
		picture: `${id}.jpg`,
		likes_count: 1,
		views_count: 1,
		...override
	};
}
