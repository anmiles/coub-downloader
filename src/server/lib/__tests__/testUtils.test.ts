import type { Coub } from '../../types';

import testUtils from '../testUtils';

const id = 'testID';

const override: Partial<Coub> = {
	test : 'value',

	/* eslint-disable camelcase */
	file_versions : {
		html5 : {
			video : {
				high : {
					url  : 'high',
					size : 5,
				},
			},
			audio : {},
		},
	},
	/* eslint-enable camelcase */
};

describe('src/lib/testUtils', () => {
	describe('createTestCoub', () => {
		it('should create test coub given id', () => {
			const testCoub = testUtils.createTestCoub(id);

			expect(testCoub).toMatchSnapshot();
		});

		it('should create test coub given id and override', () => {
			const testCoub = testUtils.createTestCoub(id, override);

			expect(testCoub).toMatchSnapshot();
		});
	});
});
