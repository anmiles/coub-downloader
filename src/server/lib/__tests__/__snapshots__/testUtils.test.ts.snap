// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/server/lib/testUtils createTestCoub should create test coub given id 1`] = `
Object {
  "file_versions": Object {
    "html5": Object {
      "audio": Object {},
      "video": Object {},
    },
  },
  "likes_count": 1,
  "media_blocks": Object {
    "external_raw_videos": Array [],
  },
  "permalink": "testID",
  "picture": "testID.jpg",
  "title": "Title: testID",
  "views_count": 1,
}
`;

exports[`src/server/lib/testUtils createTestCoub should create test coub given id and override 1`] = `
Object {
  "file_versions": Object {
    "html5": Object {
      "audio": Object {},
      "video": Object {
        "high": Object {
          "size": 5,
          "url": "high",
        },
      },
    },
  },
  "likes_count": 1,
  "media_blocks": Object {
    "external_raw_videos": Array [],
  },
  "permalink": "testID",
  "picture": "testID.jpg",
  "test": "value",
  "title": "Title: testID",
  "views_count": 1,
}
`;
